apps/glusterfs/app_topology.go

```diff
@@ -0,0 +1,110 @@
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package glusterfs
+
+import (
+	"github.com/boltdb/bolt"
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+)
+
+func (a *App) TopologyInfo() (*api.TopologyInfoResponse, error) {
+	topo := &api.TopologyInfoResponse{
+		ClusterList: make([]api.Cluster, 0),
+	}
+
+	err := a.db.View(func(tx *bolt.Tx) error {
+		clusters, err := ClusterList(tx)
+		if err != nil {
+			return err
+		}
+
+		for _, cluster := range clusters {
+			clusterInfo, err := clusterInfo(tx, cluster)
+			if err != nil {
+				return err
+			}
+			cluster := api.Cluster{
+				Id:      clusterInfo.Id,
+				Volumes: make([]api.VolumeInfoResponse, 0),
+				Nodes:   make([]api.NodeInfoResponse, 0),
+				ClusterFlags: api.ClusterFlags{
+					Block: clusterInfo.Block,
+					File:  clusterInfo.File,
+				},
+			}
+			cluster.Id = clusterInfo.Id
+
+			for _, volume := range clusterInfo.Volumes {
+				volumeInfo, err := volumeInfo(tx, volume)
+				if err != nil {
+					return err
+				}
+				if volumeInfo.Cluster == cluster.Id {
```

Comment:
I don't understand this check. Since you are iterating over volume ids on the cluster object how would you have volumes belonging to another cluster here?